File: CS/CustomMessaging.Module.Win/CustomXtraMessageBoxMessaging.cs
using System.Windows.Forms;
using DevExpress.ExpressApp;
using DevExpress.XtraEditors;
using DevExpress.ExpressApp.Win.Core;

namespace CustomMessaging.Module.Win {
    class CustomXtraMessageBoxMessaging : Messaging {
        public CustomXtraMessageBoxMessaging(XafApplication application)
            : base(application) { }
        public override DialogResult Show(
            string message, string caption, MessageBoxButtons buttons, MessageBoxIcon icon) {
            DialogResult result = XtraMessageBox.Show(
                message, caption, buttons, icon, MessageBoxDefaultButton.Button2);
            OnConfirmationDialogClosed(result);
            return result;
        }
    }
}

File: VB/CustomMessaging.Module.Win/CustomXtraMessageBoxMessaging.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.ExpressApp
Imports DevExpress.XtraEditors
Imports DevExpress.ExpressApp.Win.Core

Namespace CustomMessaging.Module.Win
    Friend Class CustomXtraMessageBoxMessaging
        Inherits Messaging
        Public Sub New(ByVal application As XafApplication)
            MyBase.New(application)
        End Sub
        Public Overrides Overloads Function Show(ByVal message As String, ByVal caption As String, ByVal buttons As MessageBoxButtons, ByVal icon As MessageBoxIcon) As DialogResult
            Dim result As DialogResult = XtraMessageBox.Show(message, caption, buttons, icon, MessageBoxDefaultButton.Button2)
            OnConfirmationDialogClosed(result)
            Return result
        End Function
    End Class
End Namespace
